src/lib/drivers/driver.js

```diff
@@ -86,6 +86,28 @@ class DriverBase {
     return Promise.reject(new Error('Not implemented'));
   }
 
+  evaluateAsync(asyncExpression) {
+    return new Promise((resolve, reject) => {
+      // Inject the call to capture inspection.
+      const expression = `window.__inspect = inspect;${asyncExpression}`;
```

Comment:
things like `axe` would still pollute `window`, but that gives tests we write entirely the chance to not leave any trace behind as well